### PR TITLE
Update Spark Operator to support ocp 4.9

### PR DESCRIPTION
**Pull request by rimolive**
Instructions to test:

- Deploy latest ODH Operator
- Create a namespace
- Add https://github.com/opendatahub-io/odh-manifests/blob/master/kfdef/kfctl_openshift.yaml to that namespace
- Spark operator should be up and running

### CODE REVIEW

File: radanalyticsio/spark/cluster/base/rolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: spark-operator-edit-resources
+  labels:
+    app: jupyterhub
+    component: spark-operator
+  namespace: "{{ meta.namespace }}"
```
Comment (LaVLaS) [inline radanalyticsio/spark/cluster/base/rolebinding.yaml:8]: This should use the kfdef parameter for the target namespace `$(namespace)`

File: radanalyticsio/spark/cluster/base/role.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: edit-resources
+  labels:
+    app: jupyterhub
+    component: spark-operator
+  namespace: "{{ meta.namespace }}"
```
Comment (LaVLaS) [inline radanalyticsio/spark/cluster/base/role.yaml:8]: iirc, you can leave the namespace out and kfctl will automatically apply the correct namespace based on where the kfdef exists

File: radanalyticsio/spark/cluster/base/role.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+kind: Role
```
Comment (LaVLaS) [inline radanalyticsio/spark/cluster/base/role.yaml:1]: I have to confirm but I think the spark-operator had clusterwide permissions

File: radanalyticsio/spark/cluster/base/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      labels:
+        component: spark-operator
+    spec:
+      serviceAccountName: spark-operator
+      containers:
+      - name: spark-operator
+        image: quay.io/opendatahub/spark-operator:1.2.0
+        env:
+        - name: WATCH_NAMESPACE
+          value: "{{ meta.namespace }}"
```
Comment (LaVLaS) [inline radanalyticsio/spark/cluster/base/deployment.yaml:28]: This variable won't work kfdef deployment.
I think the value is provided by OLM based on the OperatorGroup in the namespace
```
valueFrom:
  fieldRef:
    fieldPath: metadata.annotations['olm.targetNamespaces']
```

**Review by LaVLaS — CHANGES_REQUESTED** [review]

**Comment by nakfour** [comment]
@rimolive the CI tests are failing on the ODH dashboard, try rebasing.

File: radanalyticsio/spark/cluster/base/rolebinding.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: spark-operator-edit-resources
+  labels:
+    app: jupyterhub
+    component: spark-operator
+roleRef:
+  kind: ClusterRole
```
Comment (LaVLaS) [inline radanalyticsio/spark/cluster/base/rolebinding.yaml:9]: This should be a `Role` based on the role you are creating just for spark

File: radanalyticsio/spark/cluster/base/role.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: edit-resources
```
Comment (LaVLaS) [inline radanalyticsio/spark/cluster/base/role.yaml:4]: Can you add `spark` to the name to show that this was created for the spark operator deployment?

**Review by LaVLaS — CHANGES_REQUESTED** [review]

File: radanalyticsio/spark/cluster/base/rolebinding.yaml (hunk 5 above)
Comment (LaVLaS) [reply radanalyticsio/spark/cluster/base/rolebinding.yaml:9]: Closing this because the `Role` that is created should be a `ClusterRole`

**Review by LaVLaS — CHANGES_REQUESTED** [review]

**Comment by LaVLaS** [comment]
Closing this since the work is being completed in #509

**Pull request closed without merging**